x-pack/plugins/ml/public/explorer/kuery_filter_bar/suggestions/suggestions.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { isEmpty } from 'lodash';
+import { Suggestion } from '../suggestion';
+import { rgba } from 'polished';
+import theme from '@elastic/eui/dist/eui_theme_light.json';
+
+const List = styled.ul`
+  width: 100%;
+  border: 1px solid ${theme.euiColorLightShade};
+  border-radius: ${theme.euiSizeXs};
+  box-shadow: 0px ${theme.euiSizeXs} ${theme.euiSizeXl}
+    ${rgba(theme.euiTextColor, 0.1)};
+  position: absolute;
+  background: #fff;
+  z-index: 10;
+  left: 0;
+  max-height: ${(theme.euiSize * 20)}px;
+  overflow: scroll;
+`;
+
+class Suggestions extends Component {
+  childNodes = [];
+
+  scrollIntoView = () => {
+    const parent = this.parentNode;
+    const child = this.childNodes[this.props.index];
+
+    if (this.props.index == null || !parent || !child) {
+      return;
+    }
+
+    const scrollTop = Math.max(
+      Math.min(parent.scrollTop, child.offsetTop),
+      child.offsetTop + child.offsetHeight - parent.offsetHeight
+    );
+
+    parent.scrollTop = scrollTop;
+  };
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.index !== this.props.index) {
+      this.scrollIntoView();
+    }
+  }
+
+  render() {
+    if (!this.props.show || isEmpty(this.props.suggestions)) {
+      return null;
+    }
+
+    const suggestions = this.props.suggestions.map((suggestion, index) => {
+      const key = suggestion + '_' + index;
```

Comment:
nit, a template literal could be used here

Reply:
Done! 😄